modules/nw_multinetwork-creating-first-attachments.adoc

```diff
@@ -0,0 +1,200 @@
+// Module included in the following assemblies:
+//
+// * networking/managing_multinetworking.adoc
+
+[id='multinetwork-creating-first-attachments-{context}']
+= Creating additional network interfaces for the first time
+
+Additional interfaces for pods are defined in CNI configurations that are stored as Custom Resources.
+These custom resources can be created, listed, edited and deleted using the `oc` tool. 
+
+The procedure below configures a `macvlan` interface on a pod. This configuration may not apply to all production environments, but you use the same procedure for other CNI plugins.
+
+== Creating a CNI configuration for an additional interface as a Custom Resource
+
+[NOTE]
+====
+If you want to attach an additional interface to a pod, the Custom Resource (CR) that defines the interface must be in the same project (namespace) as the pod.
+
+====
+
+. Create a project to store CNI configurations as Custom Resources and the pods that will use the CRs.
++
+----
+$ oc new-project multinetwork-example
+$ oc project multinetwork-example
+----
+
+. Create the CR that will define an  additional network interface.
++
+----
+$ cat <<EOF | oc create -f -
+| apiVersion: "k8s.cni.cncf.io/v1"
+kind: NetworkAttachmentDefinition <2>
+metadata:
+  name: macvlan-conf <1>
+spec:
+  config: '{ <3>
+      "cniVersion": "0.3.0",
+      "type": "macvlan",
+      "master": "eth0",
+      "mode": "bridge",
+      "ipam": {
+        "type": "host-local",
+        "subnet": "192.168.1.0/24",
+        "rangeStart": "192.168.1.200",
+        "rangeEnd": "192.168.1.216",
+        "routes": [
+          { "dst": "0.0.0.0/0" }
+        ],
+        "gateway": "192.168.1.1"
+      }
+    }'
+EOF
+----
++
+<1> `name` maps to the annotation which will be used in the next step.
+<2> `kind: NetworkAttachmentDefinition`. This is the name for the custom resource this configuration will be stored. It is a custom extension of Kubernetes that defines how we attach networks to our pods.
+<3> `config`: The CNI configuration is packaged in the `config` field.
+
+The configuration above is specific to a plugin which enables `macvlan` (note the `type` line in the CNI configuration portion).
```

Comment:
plug-in, which

`macvlan`. Note the the `type` line in the CNI configuration portion.